### PR TITLE
Issue #844: Add support for 4 byte UTF-8 encoding (emoji support).

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/884.

### CODE REVIEW

File: core/includes/database/charset_converter.inc [hunk 1]
```diff
@@ -0,0 +1,256 @@
+<?php
+/**
```
Comment (serundeputy) [inline core/includes/database/charset_converter.inc:2]: @quicksketch do we need an @file docblock here?

File: core/modules/node/tests/node.test [hunk 2]
```diff
@@ -2605,3 +2605,27 @@ class NodePageCacheTest extends NodeWebTestCase {
...
+   * Tests that multi-byte UTF-8 characters are stored and retrieved correctly.
+   */
+  public function testMultiByteUtf8() {
+    $connection = Database::getConnection();
+    // On MySQL, this test will only run if 'charset' is set to 'utf8mb4' in
+    // settings.php.
+    if (!($connection->utf8mb4IsSupported() && $connection->utf8mb4IsActive())) {
+      return;
+    }
+    $title = '🐙';
```
Comment (serundeputy) [inline core/modules/node/tests/node.test:2623]: the `most` fun test eva?  :)

File: core/includes/database/charset_converter.inc (hunk 1 above)
Comment (quicksketch) [reply core/includes/database/charset_converter.inc:2]: I don't think `@file` is necessary if we only have one class in the file, as is the case here. They'd just be redundant with each other.

**Comment by quicksketch** [comment]
Merged into 1.x for 1.5.0.

**Pull request merged**